docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,116 @@
+
+ 
+### Empty Profile
+
+A device profile can begin life “empty” - meaning that it has no device resources or device commands.  Once associated to any object (device, event, reading) then, since device resource or device command additions can be made, the empty profile can gain resources or commands.  You can never remove device resources or device commands once the profile's device resources or device commands are associated to another object (event, reading).
+
+The general properties of a device profile, with the exception of name, are optional when creating an empty profile.
+
+All profiles – even empty profiles – must have name.                                                                                 
+
+### Deleting Profiles
+
+A profile can only be deleted from metadata when the profile (or its device resources or device commands) is not associated to any other EdgeX object to include device, provision watcher, event or reading.
+
+### Association Changes
+
+Today, a device profile is associated to a device via its name (the ProfileName on the Device struct).  The profile can also associated to events and readings (via the ProfileName in the Event and Reading objects).
+
+Profiles can be associated to more than one device.  Profiles can be associated to multiple events/readings.
+
+Rules about change of a device profile association have never been explicitly defined.  While the section above outline what can/cannot be changed in the profile and under what circumstances, this section outlines whether a device, event or reading can alter the associated device profile and under what circumstances.
+
+Change of the device profile on a device – never allowed unless there are no events or readings in the system associated to the device profile or its device resources or commands.
+
+A provision watcher also has an association to a device profile.  However, the device profile is only used to know the type of device to create in case of provisioning.  Therefore, if a device profile changes, it does not impact the provision watcher.
+
+A device profile cannot be deleted while it is associated to a provision watcher
+
+Change of the device profile on an event or reading is never allowed.  Events and readings are immutable.
+
+### Applicability
+
+This document dictates the rules around allowed change to device profiles (and associations).  All EdgeX services will abide by these rules (implementing appropriate checks and validation as necessary to enforce the rules) when this ADR is approved to include:
+- service APIs
+- user interfaces (like the Edge GUI)
+- client tools (like the EdgeX CLI)
+
+## Proposed Design
+
+The following metadata API changes are suggested as a means to implement the device profile rules of this ADR.
+
+- Block Device DELETE when there is any associated event/reading exists.
```

Comment:
```suggestion
- Block Device DELETE/PATCH when there is any associated event/reading exists.
```

Reply:
This ADR doesn't really mention any rules for modification of devices, so not sure `PATCH` should be included here, unless we add additional rules?

Reply:
@tonyespy , agreed and think this line should be removed for same reason. i.e. nothing to do with the profile.